source/_UIAHandler.py

```diff
@@ -344,6 +323,32 @@ def IUIAutomationNotificationEventHandler_HandleNotificationEvent(self,sender,No
 			return
 		eventHandler.queueEvent("UIA_notification",obj, notificationKind=NotificationKind, notificationProcessing=NotificationProcessing, displayString=displayString, activityId=activityId)
 
+	def _isBadUIAWindowClassName(self, windowClass):
+		"Given a windowClassName, returns True if this is a known problematic UIA implementation."
+		badUIAWindowClassNames=[
+			"SysTreeView32",
+			"WuDuiListView",
+			"ComboBox",
+			"msctls_progress32",
+			"Edit",
+			"CommonPlacesWrapperWndClass",
+			"SysMonthCal32",
+			"SUPERGRID", #Outlook 2010 message list
+			"RichEdit",
+			"RichEdit20",
+			"RICHEDIT50W",
+			"SysListView32",
+			"EXCEL7",
+			"Button",
+			# #8944: The Foxit UIA implementation is incomplete and should not be used for now.
+			"FoxitDocWnd",
+		]
+		# #7497: Windows 10 Fall Creators Update has an incomplete UIA implementation for console windows, therefore for now we should ignore it.
+		# It does not implement caret/selection, and probably has no new text events.
+		if not config.conf['UIA']['consoleUIA']:
+			badUIAWindowClassNames.append("ConsoleWindowClass")
+		return windowClass in badUIAWindowClassNames

```

Comment:
If I am reading this correctly, this will  continue adding consoleWindowClass to badUIAWindowClassNames over and over again. To avoid this, and to make the diff for this pr more readable I would suggest that badUIAWindowClassNames  stay where it originally was, and then in this function, if the window class is ConsoleWindowClass and  UIA in consoles is configured, you return True. If not, you check if windowClass is in badUIAWindowClassNames. I.e. ConsoleWindowClass is never cached.

Reply:
Could `badUIAWindowClassNames` be accessed from somewhere else?

Reply:
It could. We could address this by renaming badUIAWindowClassNames to _badUIAWindowClassNames denoting that this variable is private to the module. But I'm not sure if it is all that important.